v2/apiserver/internal/core/jobs.go

```diff
@@ -286,6 +275,42 @@ func (j *jobsService) Create(
 		}
 	}
 
+	if originalJob, ok := event.Worker.Job(job.Name); ok {
+		// If we aren't dealing with a retry event, return an error
+		if event.Labels == nil || event.Labels[RetryLabelKey] == "" {
+			return &meta.ErrConflict{
+				Type: JobKind,
+				ID:   job.Name,
+				Reason: fmt.Sprintf(
+					"Event %q already has a job named %q.",
+					eventID,
+					job.Name,
+				),
+			}
+		}
```

Comment:
This looks problematic because it during a single retry, you could get away with creating multiple jobs with the same name.

But see https://github.com/brigadecore/brigade/pull/1376#discussion_r619673271

If we used a strategy that decides up front what job results were re-usable in the retry, then most of this logic here goes away.

Reply:
Ok, logic here updated/simplified.  PTAL when convenient.